Data-Structures/Stack/Stack.js

```diff
@@ -9,45 +9,47 @@
 
 // Creates a stack constructor
 const Stack = (function () {
-  function Stack() {
-    // The top of the Stack
-    this.top = 0
-    // The array representation of the stack
-    this.stack = []
-  }
-
-  // Adds a value onto the end of the stack
-  Stack.prototype.push = function (value) {
-    this.stack[this.top] = value
-    this.top++
-  }
+  class Stack {
```

Comment:
This should be just one `Stack` class. There is no reason for this to be within a function as far as I can see.